code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -18,17 +18,20 @@
 	///How much base damage this clamp does
 	var/clamp_damage = 20
 	///Var for the chassis we are attached to, needed to access ripley contents and such
-	var/obj/vehicle/sealed/mecha/working/ripley/cargo_holder
+	var/obj/vehicle/sealed/mecha/ripley/cargo_holder
 	///Audio for using the hydraulic clamp
 	var/clampsound = 'sound/mecha/hydraulic.ogg'
 
-/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/attach(obj/vehicle/sealed/mecha/M)
+/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/attach(obj/vehicle/sealed/mecha/mecha)
 	. = ..()
-	cargo_holder = M
+	if(istype(mecha, /obj/vehicle/sealed/mecha/ripley))
+		cargo_holder = mecha
+	ADD_TRAIT(mecha, TRAIT_OREBOX_FUNCTIONAL, TRAIT_MECH_EQUIPMENT(type))
 
 /obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/detach(atom/moveto = null)
-	. = ..()
+	REMOVE_TRAIT(chassis, TRAIT_OREBOX_FUNCTIONAL, TRAIT_MECH_EQUIPMENT(type))
```

Comment:
mismatch here, you add the trait to the `mecha` and remove it from the `chassis`? Are they the same? Why not use `chassis` both times? if they're the same then it's whatever but that's confusing

Reply:
they are the same, mecha is given as an argument on attachment, chassis is the "owner" variable on the mecha itself

Reply:
for why do we not use chassis both times. what purpose do we ever gleam by saying "ooghh... mecha arg...." instead of chassis? i get that it's likely a deeper issue with mecha code but i don't get why we can't do that here if we're invoking after the parent call anyways

Reply:
yknow what ur right im just gonna use chassis both times